www/js/app.js

```diff
@@ -72,13 +66,25 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     params['showUIAnimations'] = settingsStore.getItem('showUIAnimations') ? settingsStore.getItem('showUIAnimations') === 'true' : true;
     document.getElementById('hideActiveContentWarningCheck').checked = params.hideActiveContentWarning;
     document.getElementById('showUIAnimationsCheck').checked = params.showUIAnimations;
+    // Maximum number of article titles to return (range is 5 - 50, default 25)
+    params['maxSearchResultsSize'] = settingsStore.getItem('maxSearchResultsSize') || 25;
+    document.getElementById('titleSearchRange').value = params.maxSearchResultsSize;
+    document.getElementById('titleSearchRangeVal').innerHTML = params.maxSearchResultsSize;
     // A global parameter that turns caching on or off and deletes the cache (it defaults to true unless explicitly turned off in UI)
     params['useCache'] = settingsStore.getItem('useCache') !== 'false';
     // A parameter to set the app theme and, if necessary, the CSS theme for article content (defaults to 'light')
     params['appTheme'] = settingsStore.getItem('appTheme') || 'light'; // Currently implemented: light|dark|dark_invert|dark_mwInvert
     document.getElementById('appThemeSelect').value = params.appTheme;
     uiUtil.applyAppTheme(params.appTheme);
 
+    // Define global state
+    // An array to hold searches and their state (allows modules to tell which search is current and cancel old searches)
+    state['searches'] = [{
+        'prefix': '', // A field to hold the original search string
+        'state': '',  // The state of the search: ''|'init'|'interim'|'cancelled'|'complete'
```

Comment:
I agree, it is confusing. Will think of something else.

Reply:
@mossory The only thing I could think of was to rename the global state object to "global". I think it eases comprehension, because then we have, e.g. `global.search.state`.

I did think of "mode" instead of state, but it's not very accurate. I also thought of "stage", but again "cancelled" cannot really be described as a "stage" of a search.

Reply:
@mossroy I might be a native speaker of English, but not a native speaker of JavaScript! I think `globalstate.search.status` works well. Will use it.